### PR TITLE
Create flag to skip test jobs on internal builds

### LINKED ISSUE #26162: Should have variable to disable tests via pipeline variable
There should be a way to disable the test runs when queuing a build. This could be limited to only `internal.yml` but we should have a way so testing certain non-test related pipeline changes could be done faster.

/cc @dotnet/coreclr-infra

### PR DESCRIPTION
**Pull request by AaronRobinsonMSFT**
Fixes #26162

/cc @dotnet/coreclr-infra

### CODE REVIEW

**Comment by jashook** [comment]
Note this will conflict with https://github.com/dotnet/coreclr/pull/26191

**Comment by jashook** [comment]
fyi @trylek

**Comment by AaronRobinsonMSFT** [comment]
@jashook Good call. This is a minor change by propagating the `condition` variable. @trylek this change is meant to disable _all_ test jobs (e.g. build and run). It is designed to make testing on the internal build runs quicker when we are trying to validate post-build/test validation jobs.

File: eng/test-job.yml [hunk 1]
```diff
@@ -92,9 +92,7 @@ jobs:
         - name: testhostArg
           value: 'buildtesthostonly'
 
-    # FreeBSD test jobs are disabled since we don't have any FreeBSD helix queues.
-    ${{ if eq(parameters.osGroup, 'FreeBSD') }}:
-      condition: false
+    condition: ${{ parameters.condition }}
```
Comment (elinor-fung) [inline eng/test-job.yml]: Does explicitly setting it like this override the default of only running if previous job succeeded? If a previous job failed, but `parameters.condition` is true (`_BypassTesting` is not true), will this job run?

Comment (AaronRobinsonMSFT) [reply eng/test-job.yml]: No. that is the continue on error logic which is different. Is that what you are talking about?

Comment (elinor-fung) [reply eng/test-job.yml]: Not sure? I think it is related, but not the same. From my read of the docs, `continueOnError` indicates if future jobs should run if this job fails and `condition` indicates if this job should run (where the default is only if the jobs this one depends on succeeded). The condition can be set `always()`, which suggests to me that if the condition evaluates to true, the job would run regardless of the value of `continueOnError` in previous jobs?

Comment (jkoritzinsky) [reply eng/test-job.yml]: We might need to do something here like `and(succeeded(), ${{ parameters.condition }})` to make sure we only run the job when there are no failed jobs.

**Comment by trylek** [comment]
Thanks @AaronRobinsonMSFT for the explanation. For now I'd expect your change to go in first (especially as I'm still somewhat concerned about the stability impact of unconditionally uploading the test artifacts to Azure) and I'll rebase and fix my change before updating my PR.

File: eng/build-job.yml [hunk 2]
```diff
@@ -22,10 +22,7 @@ jobs:
     name: ${{ format('build_{0}_{1}_{2}', parameters.osIdentifier, parameters.archType, parameters.buildConfig) }}
     displayName: ${{ format('Build {0} {1} {2}', parameters.osIdentifier, parameters.archType, parameters.buildConfig) }}
 
-    # FreeBSD builds are disabled in the public project because we
-    # don't have any FreeBSD agents in the public pool.
-    ${{ if and(eq(parameters.osGroup, 'FreeBSD'), eq(variables['System.TeamProject'], 'public')) }}:
-      condition: false
+    condition: ${{ parameters.condition }}
```
Comment (elinor-fung) [inline eng/build-job.yml]: Is this needed? `condition' isn't listed as parameter at the top and nothing seems to explicitly set this when using `build-job`?

Comment (AaronRobinsonMSFT) [reply eng/build-job.yml]: You're right. I think this was me simply removing the FreeBSD settings. I am going to remove this entirely.

**Pull request merged**